packages/block-library/src/navigation/index.php

```diff
@@ -57,23 +63,22 @@ private static function is_responsive( $attributes ) {
 	/**
 	 * Returns whether or not a navigation has a submenu.
 	 *
-	 * @param WP_Block_List $inner_blocks The list of inner blocks.
-	 * @return bool Returns whether or not a navigation has a submenu.
+	 * @param array $inner_blocks An array of inner blocks.
+	 * @return bool Returns whether or not a navigation has a submenu and also sets the member variable.
 	 */
 	private static function has_submenus( $inner_blocks ) {
+		if ( true === static::$has_submenus ) {
+			return static::$has_submenus;
+		}
+
 		foreach ( $inner_blocks as $inner_block ) {
-			$inner_block_content = $inner_block->render();
-			$p                   = new WP_HTML_Tag_Processor( $inner_block_content );
-			if ( $p->next_tag(
-				array(
-					'name'       => 'LI',
-					'class_name' => 'has-child',
-				)
-			) ) {
-				return true;
+			if ( 'core/navigation-submenu' === $inner_block->name ) {
+				static::$has_submenus = true;
+				return static::$has_submenus;
```

Comment:
would rhe function be clearer if we used `break` here so we have one early return and one final return? Seems a nit but the last return is always `false`in this procedure.